### DISCUSSION TITLE
Parametric plot - time of a point

### DISCUSSION THREAD

**Discussion by s-celles** [OpenModelica/OpenModelica#11196] [Q&A] [answered]
Hello,

I am studying the motorization of the trolley of a boat loading gantry.

The gantry consists of a frame that moves on rails on the port quay.
The frame supports an M=32 ton overhead crane carriage moving perpendicular to the quay.

The truck can lift or lower loads of mass m=50 tonnes.

From a mechanical point of view we are in the presence of three independent translation movements, along the three axes. However, in the problem, we are only interested in the movement of the cart (along the x axis).

[![figure][1]][1]

The kinematic profile (speed profile) of the carriage movement is given below:

[![speed_profile][2]][2]

The movement is ensured by a motor driving the wheels of the cart by a reduction gear. The motor runs at 1450 rpm when the carriage moves at 3 m/s. The torque of the machine is constant during the acceleration and deceleration phase. The efficiency of the wheels is 100%.

The efficiency of the reducer is η = 80% in both directions of energy transfer:
- From the engine to the wheels: in the traction phase, driven load
- From the wheels towards the engine: during braking, driving load.

I did a Modelica class to simulate that.

It looks like:

[![enter image description here][3]][3]

```modelica
model chariot_portique
  Modelica.Mechanics.Rotational.Components.IdealGearR2T idealGearR2T(ratio = 50.61)  annotation(
    Placement(visible = true, transformation(origin = {74, -36}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
  Modelica.Mechanics.Rotational.Sensors.SpeedSensor speedSensor1 annotation(
    Placement(visible = true, transformation(origin = {56, 14}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
  Modelica.Mechanics.Rotational.Components.LossyGear lossyGear(lossTable = [0, 0.8, 0.8, 0, 0], ratio = 1)  annotation(
    Placement(visible = true, transformation(origin = {18, -36}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
  Modelica.Mechanics.Translational.Sensors.SpeedSensor speedSensor annotation(
    Placement(visible = true, transformation(origin = {122, 20}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
  Modelica.Mechanics.Translational.Components.Mass mass(m = 82000)  annotation(
    Placement(visible = true, transformation(origin = {146, -36}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
  Modelica.Blocks.Sources.CombiTimeTable combiTimeTable(extrapolation = Modelica.Blocks.Types.Extrapolation.HoldLastPoint, table = [0, 0; 0, 1520; 4, 1520; 4, 0; 8, 0; 8, -974; 12, -974; 12, 0; 16, 0; 16, -1520; 20, -1520; 20, 0; 24, 0; 24, 974; 28, 974; 28, 0; 32, 0], tableOnFile = false) annotation(
    Placement(visible = true, transformation(origin = {-6, -66}, extent = {{-80, 20}, {-60, 40}}, rotation = 0)));
  Modelica.Mechanics.Rotational.Sources.Torque torque annotation(
    Placement(visible = true, transformation(origin = {-28, -36}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
  Modelica.Mechanics.Translational.Sensors.ForceSensor forceSensor annotation(
    Placement(visible = true, transformation(origin = {110, -36}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
equation
  connect(idealGearR2T.flangeR, speedSensor1.flange) annotation(
    Line(points = {{64, -36}, {46, -36}, {46, 14}}));
  connect(lossyGear.flange_b, idealGearR2T.flangeR) annotation(
    Line(points = {{28, -36}, {64, -36}}));
  connect(speedSensor.flange, idealGearR2T.flangeT) annotation(
    Line(points = {{112, 20}, {84, 20}, {84, -36}}, color = {0, 127, 0}));
  connect(torque.flange, lossyGear.flange_a) annotation(
    Line(points = {{-18, -36}, {8, -36}}));
  connect(combiTimeTable.y[1], torque.tau) annotation(
    Line(points = {{-65, -36}, {-40, -36}}, color = {0, 0, 127}));
  connect(forceSensor.flange_a, idealGearR2T.flangeT) annotation(
    Line(points = {{100, -36}, {84, -36}}, color = {0, 127, 0}));
  connect(forceSensor.flange_b, mass.flange_a) annotation(
    Line(points = {{120, -36}, {136, -36}}, color = {0, 127, 0}));
  annotation(
    experiment(StopTime = 32),
    uses(Modelica(version = "4.0.0")),
  Diagram(coordinateSystem(extent = {{-200, -100}, {200, 100}})),
  Icon(coordinateSystem(extent = {{-200, -100}, {200, 100}})),
  version = "");
end chariot_portique;
```

I can plot torque vs time or rotational speed vs time like so:

[![plot_vs_time][4]][4]

I want to plot parametric plot like torque (y-axis) vs rotational speed (x-axis) ie something like

[![xy_plot][5]][5]

I tried the second icon [![icons][6]][6]

and found that I can do this kind of plot (using shift to select first variable for x-axis and without using shift to select the second variable for y-axis).

I get :

[![xy_plot_sim][7]][7]

Which is exactlely what I was expecting.

Unfortunately when mouse over plot it only displays Value: 1.52 at 0.742837

[![values][8]][8]

Is there a way to know the time of this point without changing plot ?

By changing plot we can find time for such a point ie 2.048s : 
![image](https://github.com/OpenModelica/OpenModelica/assets/109167/2902b70e-695a-441f-aef2-eb0d2fbef54f)

Ideally we could also move point across time using arrows keys or an icon.

  [1]: https://i.stack.imgur.com/BN9sj.png
  [2]: https://i.stack.imgur.com/b8Sew.png
  [3]: https://i.stack.imgur.com/iPhOv.png
  [4]: https://i.stack.imgur.com/Attdd.png
  [5]: https://i.stack.imgur.com/FD45T.png
  [6]: https://i.stack.imgur.com/4vqrm.png
  [7]: https://i.stack.imgur.com/knxXJ.png
  [8]: https://i.stack.imgur.com/290WX.png

**Comment by s-celles** [comment]
Maybe a kind of interaction with the variables explorer (or navigator) could be considered

![image](https://github.com/OpenModelica/OpenModelica/assets/109167/947fdd2b-429d-4b1c-8f0b-39e995e55b2c)

in both direction from plot to variables explorer ie left clicking on the plot could change Simulation Time Unit.

or for variable explorer to plot ("playing" simulation could move point on plot)

Any opinion about that ?

**Comment by casella** [comment, marked as answer]
Both suggestions are fine. Do you mind opening two tickets, one for each feature?

**Comment by s-celles** [comment]
Two tickets have been opened.

https://github.com/OpenModelica/OpenModelica/issues/11198
https://github.com/OpenModelica/OpenModelica/issues/11199

Thanks

**Comment by casella** [comment]
OK. We'll consider them for the next release.
